packages/labs/ssr/src/lib/render-value.ts

```diff
@@ -620,7 +695,21 @@ function* renderTemplateResult(
         break;
       case 'child-part': {
         const value = result.values[partIndex++];
-        yield* renderValue(value, renderInfo);
+        let isValueHydratable = hydratable;
+        if (isTemplateResult(value)) {
+          isValueHydratable = isHydratable(value);
+          if (!isValueHydratable && hydratable) {
+            throw new Error(
+              `A server-only template can't be rendered inside an ordinary, hydratable template. The outer template was:
```

Comment:
Great errors! May be easier to state that a server-only template can only be rendered at the top level or within other server-only templates.

Reply:
Added that as another phrasing to this error message.

Reply:
I can see this causing some headaches for us down the road. Would this be a valid workaround if I have a server-only template that I want to use inside a normal Lit template?

```
const normalChildTemplate = html`<h1>Hello World</h1>`
const serverChildTemplate = serverHtml`<span>${normalChildTemplate}</span>`

const rootTemplate = html`<div>${normalChildTemplate}</div>`
const otherServerTempate = serverHtml`<p>${serverChildTemplate}</p>`
```

This seems like it would work as a mechanism for sharing templates? I ask because being able to share templates between client and server is one of the big draws of Lit SSR for us (at Reddit).